checkRange.c

```diff
@@ -0,0 +1,30 @@
+#include "header.h"
+
+int count=0;
+
+int isChargeNotInRange(float chargeRate)
+{
+  if(chargeRate > 0.8)
+  {
+    count++;
+  }
+  return count;
+}
+
+int isStateNotInRange(float soc)
+{
+    if(soc < 20 || soc > 80) 
+    {
+      count++;
+    }
+  return count;
+}
+
+int isTempNotInRange(float temperature)
+{
+    if(temperature < 0 || temperature > 45) 
```

Comment:
Avoid repeated < || > checks by having them in a function e.g., `isWithinRange`.
This way only that function needs to be tested and re-used everywhere.